File: ImageFlow/ocaml/bridge/nsnumber.c
#include <objc/objc.h>
#include <objc/objc-runtime.h>

#include <caml/mlvalues.h>
#include <caml/memory.h>

#include "objc.h"

static id NSNumberClass = NULL;

CAMLprim value NSNumber__numberWithDouble(value theDouble) {
  CAMLparam1(theDouble);
  static SEL sel = NULL;
  if (sel == NULL)
    sel = sel_registerName("numberWithDouble:");
  if (NSNumberClass == NULL)
    NSNumberClass = objc_getClass("NSNumber");
  CAMLreturn(objc_wrap(objc_msgSend(NSNumberClass, sel, Double_val(theDouble))));
}

File: ImageFlow/ocaml/bridge/nsnumber.m
#import <caml/mlvalues.h>
#import <caml/memory.h>

#import <Foundation/Foundation.h>

#import "objc.h"

CAMLprim value NSNumber__numberWithDouble(value theDouble) {
  CAMLparam1(theDouble);
  CAMLreturn(objc_wrap([NSNumber numberWithDouble:Double_val(theDouble)]));
}

CAMLprim value NSNumber__numberWithInt(value theInt) {
  CAMLparam1(theInt);
  CAMLreturn(objc_wrap([NSNumber numberWithInt:Int_val(theInt)]));
}
